### ISSUE TITLE
Reduce the number of dispatcher scale up/down events

### ISSUE THREAD

**Issue by AlanCoding** [ansible/awx#12777]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Feature type

Enhancement to Existing Feature

### Feature Summary

The task dispatcher will scale workers down at up at a rate much higher than necessary.

Expectation is that if we have a constant number of jobs running, scale ups and downs should not be necessary _at all_. Meaning, if we have 0 jobs running, we should always be fine at the minimum number of workers. But if we have 10 jobs running, we should also have no need to scale up or down, if that number has held constant for more than ~30 seconds.

Right now this is not the case. We know of cases when we may actually scale a worker _down and then up_ right before starting a new task.

### Select the relevant components

- [ ] UI
- [X] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [ ] Other

### Steps to reproduce

launch a constant number of sleep tasks, observe number of workers.

this will be easier to do when https://github.com/ansible/awx/issues/12776 is completed.

### Current results

We know from the logs that workers are always scaling up and down, even when there is no need to.

### Sugested feature result

Smarter worker scaling logic.

### Additional information

I attempted this in https://github.com/ansible/awx/pull/12584 and reverted that in https://github.com/ansible/awx/pull/12766

In the end, that approach left _too many_ idle workers in reserve. We need a solution that keeps _just enough_ idle workers free. In practice, this probably means keeping as many workers as the number of scheduled tasks that might run at the same time. So, I suggest that the best solution would be to add a timeout to scheduled tasks, then compute the max number of simultaneous scheduled tasks, and then keep this number of workers available _beyond the current workers running non-scheduled tasks_. This is a much more complex heuristic than my prior attempt, but it would give a much more satisfactory result. I'm unhappy with both the increased memory use of my prior attempt, and the impact of database connection usage.
